pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFieldSpec.java

```diff
@@ -119,6 +137,15 @@ public DateTimeFormatSpec getFormatSpec() {
       formatSpec = new DateTimeFormatSpec(_format);
       _formatSpec = formatSpec;
     }
+
+    if (_sampleValue != null) {
```

Comment:
We might not want to validate the sample value here. Suggest adding a method `validateSampleValue()` and call it when uploading the schema.